service/src/test/java/com/memento/service/impl/CityServiceImplTest.java

```diff
@@ -0,0 +1,62 @@
+package com.memento.service.impl;
+
+import com.memento.model.City;
+import com.memento.repository.CityRepository;
+import com.memento.shared.exception.ResourceNotFoundException;
+import org.junit.Test;
+import org.junit.runner.RunWith;
+import org.mockito.InjectMocks;
+import org.mockito.Mock;
+import org.mockito.junit.MockitoJUnitRunner;
+
+import java.util.Collections;
+import java.util.Optional;
+
+import static org.mockito.Mockito.*;
+
+@RunWith(MockitoJUnitRunner.class)
+public class CityServiceImplTest {
+
+    @Mock
+    private CityRepository cityRepository;
+
+    @InjectMocks
+    private CityServiceImpl cityService;
+
+    @Test
+    public void getAllTest() {
+        when(cityRepository.findAll()).thenReturn(Collections.emptyList());
+        cityService.getAll();
+        verify(cityRepository).findAll();
+    }
+
+    @Test
+    public void saveTest() {
+        City city = mock(City.class);
+        when(cityRepository.save(any(City.class))).thenReturn(city);
+        cityService.save(city);
+        verify(cityRepository).save(any(City.class));
+    }
+
+    @Test(expected = ResourceNotFoundException.class)
+    public void updateTest() {
```

Comment:
All that code for just to take ResourceNotFoundException ? I'm sure there is better approach for this.